File: java/src/test/java/problems301_400/RansomNoteTest.java
package problems301_400;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class RansomNoteTest extends SampledTest {
    @Test
    public void test() {
        for (RansomNote solution : RansomNote.values()) {
            for (RansomNoteSample sample : getSamples(RansomNoteSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.canConstruct(sample.input.ransomNote, sample.input.magazine))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems301_400/RansomNoteTest.kt
package problems301_400

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class RansomNoteTest : SampledTest() {
    @Test
    fun test() =
        RansomNote.entries.forEach { solution ->
            getSamples<Array<RansomNoteSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.canConstruct(input.ransomNote, input.magazine))
                    .isEqualTo(output)
            }
        }
}
